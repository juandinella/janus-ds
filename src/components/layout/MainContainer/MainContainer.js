import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import styles from './MainContainer.styles'


/**
 * MainContainer
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledMainContainer = styled.section.withConfig({shouldForwardProp})`${styles}`

const MainContainer = ({ id, children }) => {
  return <StyledMainContainer id={id}>{children}</StyledMainContainer>
}

MainContainer.displayName = 'MainContainer'

MainContainer.propTypes = {
  id: PropTypes.string,
  children: PropTypes.node.isRequired,
}

MainContainer.defaultProps = {}

export default MainContainer
