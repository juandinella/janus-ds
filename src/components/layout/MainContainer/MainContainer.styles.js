import { css } from 'styled-components'

export default css`
  width: 400px;
  min-height: 100px;
  border: none;
  position: fixed;
  height: 85vh;
  z-index: 999999;
  left: unset;
  bottom: 1rem;
  right: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
`
