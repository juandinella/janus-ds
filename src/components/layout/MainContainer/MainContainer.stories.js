import MainContainer from './MainContainer'

import {
  getTemplate,
} from '../../../helpers/storybook'

import Header from '../../molecules/Header'
import MessageInput from '../../molecules/MessageInput'
import HeaderContainer from '../HeaderContainer'
import MessageInputContainer from '../MessageInputContainer'
import MessagesContainer from '../MessagesContainer'
const Template = getTemplate(MainContainer)

export default {
  title: 'Layout/MainContainer',
  component: MainContainer,
  args: {
    children: (
      <>
        <HeaderContainer>
          <Header
            isOnline
            lastSeen="Active now"
            username="Juan Di Nella"
          />
        </HeaderContainer>
        <MessagesContainer />
        <MessageInputContainer>
          <MessageInput />
        </MessageInputContainer>
      </>
    )
  },
  argTypes: {},
  parameters: { __sb: { h: '500px' } },
}

export const Default = Template.bind({})
