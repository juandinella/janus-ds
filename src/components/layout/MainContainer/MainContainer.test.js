import initStoryshots from '@storybook/addon-storyshots'
import { cleanup, render, screen } from '@testing-library/react'
import React from 'react'
import MainContainer from './MainContainer'

initStoryshots({
  storyKindRegex: /^Layout\/MainContainer$/,
})

const componentProps = {
  id: 'container',
  children: <span data-testid="container_span">Im a span</span>,
}

afterEach(cleanup)

describe('<MainContainer />', () => {
  test('should render component correctly', () => {
    render(<MainContainer {...componentProps} />)
    const container = screen.getByTestId(componentProps.id)
    expect(container).toBeInTheDocument()
  })

  test('should render children correctly', () => {
    const { queryByTestId } = render(<MainContainer {...componentProps} />)
    expect(queryByTestId(componentProps.id).children).toHaveLength(1)
    expect(queryByTestId('container_span')).toBeTruthy()
    expect(queryByTestId('container_span').textContent).toEqual('Im a span')
  })
})
