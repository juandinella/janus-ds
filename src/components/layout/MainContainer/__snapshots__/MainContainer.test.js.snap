// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout/MainContainer Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <div
    className="sc-1bywnpk-0 bvrKrw"
  />
</div>
`;
